### ISSUE TITLE
Reactions: no visibility over whether I have already left a reaction + see, who has reacted

### ISSUE THREAD

**Issue by oksks** [citizenos/citizenos-fe-old#470]
**What is the problem?**
1) If I'm not mistaken, it used to be so that when I left a reaction on a reply or argument, the thumbs-up (or down) icon was coloured red or green, while other users' reactions have been shown as grey. 
Now it gets a grey colour just like everyone else's, so there's no way to find out whether I've left a reaction or not without clicking it again and seeing if the number changes:

(on the following image, I've voted "pro" for this argument)
<img width="331" alt="Screenshot 2020-04-30 at 13 38 21" src="https://user-images.githubusercontent.com/34126353/80701328-e4449600-8ae7-11ea-8b17-f9a8f2ac09f2.png">

2) I can't see who's already left a reaction - the only clickable action is to react.

**Why is this a problem?**
1) Upon revisiting topics, I can no longer remember whether I've reacted to something or not (it looks like I haven't), and without a visual cue, I keep having to re-read arguments and replies and reacting all over again in all my topics, which is waste of time. 
2) I want to see what others think - do they agree or disagree with something, but there is no way to do that (maybe only via scrolling through a humongous Activity Feed to search for reactions)

**Possible fix:**
1) Colour my reaction if I've voted
2) Clicking the number of reactions should show who's reacted

**Comment by loorm** [comment]
Successfully replicated. Scheduling for triage 21.

**Comment by loorm** [comment]
Triage 21. @kevincrepin - tiny input needed from you. What is the appropriate visual indication, that user has already reacted to this.
Est dev time - 8 hours.
Prioritizing.

**Comment by kevincrepin** [comment]
Are we talking only about the thumbs up/down here or also about the replies to an argument?

About thumbs up/down: Agree it's unclear now, let's do it so that if you click on up, it turns green and down, it turns red. This way you can at least see which you have selected.

_Clicking the number of reactions should show who's reacted_ -> Which number is this, the "show replies (x)"?

On a small scale we can make some improvements to indicate better if you have previously interacted with an argument, but this doesn't eliminate the need of going through the entire arguments section and finding those places where you interacted.

On bigger scale there is a need for a separate place where I can better manage my interactions with the "arguments" component, ie. just like I have "my topics" and "my groups" there can be a "my arguments" which holds all my interactions on this level of a topic. It includes my own arguments and replies to other people's arguments. I can search, filter and order and from there also jump into the actual topic.

**Comment by kevincrepin** [comment]
Example for "quick fix":
![Screenshot 2020-05-11 at 12 42 07](https://user-images.githubusercontent.com/26792281/81547698-e2e75900-9384-11ea-930f-a82e0f23e366.png)

If possible let's update the argument green to have better contrast as well.

**Comment by oksks** [comment]
> Clicking the number of reactions should show who's reacted -> Which number is this, the "show replies (x)"?

I meant these numbers:

<img width="438" alt="Screenshot 2020-05-11 at 13 01 27" src="https://user-images.githubusercontent.com/34126353/81549556-96514d00-9387-11ea-8638-a93249a54251.png">

Clicking on either of these could maybe pop up a list of all users who've voted, with their actual votes (green or red thumbs up) next to their name or avatar.

**Comment by kevincrepin** [comment]
It's a bit difficult because now you can click on it to give the thumbs up/down. So that would mean that we'd have 2 separate click areas. Click on icon and it casts the vote. Click on number and see the list. Not easily understandable interaction.

Also wondering if it's ok to show these votes? "Ethical"? It goes from being a low commitment interaction to a high commitment if people can actually see who voted how.

**Comment by oksks** [comment]
@kevincrepin I feel like in this era, Facebook, Instagram and the likes have trained the internet population to click on the number of reactions to see who's reacted and how :) 

> Also wondering if it's ok to show these votes?

Thumbsups and thumbdowns are reactions (instead of votes), and they are public already via Activity Feed:
<img width="764" alt="Screenshot 2020-05-14 at 00 54 34" src="https://user-images.githubusercontent.com/34126353/81869952-a03ca100-957d-11ea-95da-3b47accbad55.png">
Votes should definitely stay private.

**Comment by kevincrepin** [comment]
Left a reaction "up" is so very weird :)

I have to make some changes to the UI to make it a bit more understandable that the icon and the number are separately clickable. Not so easy because we also don't use underline here to reduce the visual noise.

So maybe the icon can be more substantial and the number can have text, making it easier for both to be clicked on. If you have already clicked on it then the icon gets coloured and the text becomes bold.

![Screenshot 2020-05-15 at 11 56 16](https://user-images.githubusercontent.com/26792281/82032074-68c81480-96a3-11ea-888b-ce3d3a1343d1.png)

Clicking on the text could open a dialog with the list of people and their reactions. I would use something other than "up" because, as we see from the activity feed, out of context it doesn't make much sense.

![Screenshot 2020-05-15 at 11 55 43](https://user-images.githubusercontent.com/26792281/82032221-96ad5900-96a3-11ea-98f9-64dee4a64719.png)

**Comment by oksks** [comment]
> Left a reaction "up" is so very weird :)

agree - wdyt, maybe rewording it to "thumbs up" would be an easy fix? 
on another thought, that's a fix for English only - in other languages the whole concept breaks up pretty quickly, as there's usually no specific phrase/word for "thumbs up/down" and it'll have to be translated as "approve", "like" or similar, and it'll be quite hard to hit that right tone :( 
any ideas how to tackle it via copy @kevincrepin?

=========

regarding number with text next to them, from localization POV that's not ideal, as in many languages the translation of the text (and sometimes, even the words in use) will be dependent on the number (and noun's gender) - for example, here's Russian translation for "X people":

- 1 пользовате**ль**
- 2, 3, 4 пользовате**ля**
- 5-19 пользовате**лей**

etc.

**Comment by kevincrepin** [comment]
Yes true, also just the word "people" becomes potentially too long in other languages...

Maybe let's just have the number then. We can see over time if people realise whether they can click on it or not and if they don't, do they miss this information.

![Screenshot 2020-05-18 at 11 32 06](https://user-images.githubusercontent.com/26792281/82191681-8a233d80-98fb-11ea-9839-7b4abd2c58f1.png)

About wording: A bit difficult yeah, but maybe makes sense to also for this look at the bigger players and take over their wording, ie. like/dislike...

**Comment by loorm** [comment]
I would lean towards agree / disagree. We're not social media and "likes" have receive their fair amount of criticism. We're a decision-making app, therefore agree / disagree is much more context-relevant.

@kevincrepin - otherwise, are we ready for devs to implement this?

**Comment by kevincrepin** [comment]
ready here: https://projects.invisionapp.com/d/main#/console/9829159/418248426/inspect

**Comment by KatiVellak** [comment]
I agree with Margo on "I would lean towards agree / disagree. We're not social media and "likes" have receive their fair amount of criticism. We're a decision-making app, therefore agree / disagree is much more context-relevant."
Otherwise, no further comments, legally reviewed.

**Comment by ilmartyrk** [comment]
@kevincrepin can you please add icon files too :)

**Comment by kevincrepin** [comment]
@ilmartyrk should be available now

**Comment by ilmartyrk** [comment]
@kevincrepin that was quick, thanks!

**Comment by loorm** [comment]
Tested. Confirmed working.
